### ISSUE TITLE
Ensure that `::` etc are consistent

### ISSUE THREAD

**Issue by paulbutcher** [flix/flix#4708]
If the programmer re-uses `Cons`, this causes the behaviour of `::` to change:

```
namespace A {

    enum MyType { case Cons(Int32, MyType) }

    pub def foo(): List[Int32] = 1 :: Nil
}
```

```
❌ -- Type Error -------------------------------------------------- file:///Users/paulbutcher/Projects/Flix/flix-playground/src/Main.flix

>> Unable to unify the types: 'MyType' and 'List[t13187821]'.

5 |     pub def foo(): List[Int32] = 1 :: Nil
                                       ^^
                                       mismatched types.

Type One: ((Int32, MyType)) -> MyType
Type Two: ((Int32, List[t13187821])) -> t13167268
```

We should restrict `::` to always use `List.Cons`. Similarly `:::`, `Set#{}`, `Map#{}` (any others?)

See also #1505.

**Comment by magnus-madsen** [comment]
In other words:

- `::` must be `List.Cons`
- `:::` must be `List.append`.

I think set and map are restricted, but we should double check.

All this is in the Weeder.

**Comment by magnus-madsen** [comment]
https://github.com/flix/flix/blob/master/main/src/ca/uwaterloo/flix/language/phase/Weeder.scala#L1325

**Comment by magnus-madsen** [comment]
https://github.com/flix/flix/blob/master/main/src/ca/uwaterloo/flix/language/phase/Weeder.scala#L2201

**Comment by paulbutcher** [comment]
Fixed via #4815
